### PR TITLE
Add an optional dependency on memtrace to profile memory in coqc.

### PR DESCRIPTION
**Pull request by ppedrot**
I grew tired of having to cherry-pick a memtrace commit every single time I wanted to profile something, and have to recompile all the libraries.

(Hopefully this will work, my dune-fu is not great.)

### CODE REVIEW

**Comment by ppedrot** [comment]
cc @ejgallego

File: toplevel/dune [hunk 1]
```diff
@@ -5,7 +5,7 @@
  (wrapped false)
  ; until ocaml/dune#4892 fixed
  ; (private_modules g_toplevel)
- (libraries coq-core.stm))
+ (libraries coq-core.stm (select memtrace_init.ml from (memtrace -> memtrace_init.memtrace.ml) (!memtrace -> memtrace_init.default.ml))))
```
Comment (Alizter) [inline toplevel/dune:8]:
```suggestion
 (libraries coq-core.stm
  (select memtrace_init.ml from
   (memtrace -> memtrace_init.memtrace.ml)
   (!memtrace -> memtrace_init.default.ml))))
```

**Comment by ppedrot** [comment]
Should we discuss this at the weekly meeting?

**Comment by ejgallego** [comment]
See also https://github.com/coq/coq/issues/13161

**Comment by ejgallego** [comment]
Looks good to me @ppedrot , only question is if you want to make this a configure-time option instead of the PR default of unconditionally initializing memtrace when installed (that seems dangerous for users, doesn't it?)

**Comment by ppedrot** [comment]
This only initializes memtrace when the MEMTRACE environment variable is set, otherwise it's a no-op. Not sure about the security implications, though, the main issue when unconditionally linking it is the size of binaries as per the -linkall flag.

**Comment by SkySkimmer** [comment]
So should we merge or what?

**Review by ejgallego — APPROVED** [review]
LGTM

**Comment by ejgallego** [comment]
I heard somewhere that `select` is kind of a weird feature and will be phased out eventually, but I'm unsure what is the alternative, this should work fine for a few years at least.

**Comment by Alizter** [comment]
@ejgallego It's definitely an awkward feature, but I don't think there are any plans to replace it at the moment. A relevant issue is https://github.com/ocaml/dune/issues/397. For the time being, it is fine to keep.

**Comment by ejgallego** [comment]
@coqbot: merge now

**Pull request merged**